.github/workflows/check-build-deploy.yaml

```
@@ -44,7 +44,7 @@ jobs:
                 enable-cache: true
 
             - name: Install mypy From Locked Dependencies
-              run: uv sync --no-group dev --group type-check
+              run: uv sync --no-group dev --group type-check --group test
```

Why is the `test` dependency group required for the type-checking job?

---

type checking will fail on #516 if the gitpython module is not installed, which is part of the testing dependency group

---

If there is a dependency required for the type-checking group & the testing group, then it should be split into a new group that is included by both of those groups. In this case however, it seems like it is used at runtime and should thus be part of the `main` group, as runtime would fail if it is only available to type checking.

```suggestion
              run: uv sync --no-group dev --group type-check
```